### ISSUE TITLE
require-param rule destructuring false errors

### ISSUE THREAD

**Issue by jaydenseric** [gajus/eslint-plugin-jsdoc#569]
The `require-param` rule option `checkTypesPattern` currently doesn't work deeply for destructured parameters.

## Expected behavior

If you use `object` for an `@param` root type but then a named type (e.g. `FooBar`) for one of it's destructured properties, there should be no `jsdoc/require-param` lint errors.

## Actual behavior

There are unexpected lint errors.

## ESLint Config

```json
{
  "rules": {
    "jsdoc/require-param": "error"
  }
}
```

## ESLint sample

This has the unexpected lint error `Missing JSDoc @param "options.a.b" declaration. eslint(jsdoc/require-param)`:

```js
/**
 * Description.
 * @param {object} options Options.
 * @param {FooBar} options.a A description.
 */
function foo({ a: { b } }) {}
```

This does not:

```js
/**
 * Description.
 * @param {Foo} options Options.
 * @param {FooBar} options.a A description.
 */
function foo({ a: { b } }) {}
```

## Environment

* Node version: v14.4.0
* ESLint version v7.2.0
* `eslint-plugin-jsdoc` version: v27.0.4

**Comment by brettz9** [comment]
For a workaround, you might try something

```js
/**
 * @typedef {object} FooOptionsObject
 * @property {FooBar} a A description.
*/

/**
 * Description.
 * @param {FooOptionsObject} options Options.
 */
function foo({ a: { b } }) {}
```

But yeah, we need to fix this, hopefully along with #540.
